src/pipeline.ts

```diff
@@ -179,6 +200,25 @@ export class GitHubWorkflow extends PipelineBase {
     this.publishAssetsAuthRegion = props.publishAssetsAuthRegion ?? 'us-west-2';
   }
 
+  /**
+   * Deploy a single Stage by itself with options for further GitHub configuration.
+   *
+   * Add a Stage to the pipeline, to be deployed in sequence with other Stages added to the pipeline.
+   * All Stacks in the stage will be deployed in an order automatically determined by their relative dependencies.
+   */
+  public addStageWithGitHubOpts(stage: Stage, options?: AddGitHubStageOpts): StageDeployment {
```

Comment:
can't override the `addStage` API because of this JSII error: 

```bash
[2022-04-25T13:41:01.386] [ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created
error JSII5006: "cdk-pipelines-github.GitHubWorkflow#addStage" changes the type of parameter "options" to cdk-pipelines-github.AddGitHubStageOpts when overriding aws-cdk-lib.pipelines.PipelineBase. Change it to aws-cdk-lib.pipelines.AddStageOpts
```

Reply:
`Opts` -> `Options` I think?